overlord/configstate/config/transaction.go

```diff
@@ -361,3 +457,44 @@ func (e *NoOptionError) Error() string {
 	}
 	return fmt.Sprintf("snap %q has no %q configuration option", e.SnapName, e.Key)
 }
+
+// VirtualCfgFunc can be used for virtual "transaction.Get()" calls
+type VirtualCfgFunc func(snapName, key string, result VirtualResult) error
+
+// VirtualResult is used to store the result of a VirtualCfgFunc call
+type VirtualResult struct {
+	res interface{}
+}
+
+func (vcr *VirtualResult) Set(val interface{}) {
+	rv := reflect.ValueOf(vcr.res)
+	rv.Elem().Set(reflect.ValueOf(val))
+}
+
+// virtualMap contain "virtual" configuration keys like
+// e.g. "core.hostname".  The data is never stored in the state but
+// instead directly get/set.
```

Comment:
I think the docstring should explain that we map snap names to config keys (dotted paths really, right?).

Reply:
Thanks, good point. I updated this now.